### ISSUE TITLE
[Paid] [$250]  Android - Wallet - "Get physical card" button overlaps with device navigation bar

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#53650]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.72-0
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Yes
**If this was caught during regression testing, add the test name, ID and link from TestRail:** Exp
**Email or phone of affected tester (no customers):** applausetester+kh1411002@applause.expensifail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
Precondition:
- Device navigation is buttons and not gestures.
- User has been assigned a physical Expensify card.

1. Launch ND or hybrid app.
2. Go to Wallet.
3. Tap on the physical card.

## Expected Result:
"Get physical card" button will not overlap with device navigation bar.

## Actual Result:
"Get physical card" button overlaps with device navigation bar.

## Workaround:
Unknown

## Platforms:
- [x] Android: Standalone
- [x] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/b5528f94-10ce-4f08-a4ff-88ceaf4dafb1

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021866268126921141487</li>
        <li>Upwork Job ID: 1866268126921141487</li>
        <li>Last Price Increase: 2024-12-09</li>
<li>Automatic offers: </li>
<ul>
<li>FitseTLT | Contributor | 105374034</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @mananjadhav</details>

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
"Get physical card" button overlaps with device navigation bar.

### What is the root cause of that problem?

In `ExpensifyCardPage` component we are setting `includeSafeAreaPaddingBottom` as false 
https://github.com/Expensify/App/blob/24793797a140b3f65a7de0511adccdcb51bc1394/src/pages/settings/Wallet/ExpensifyCardPage.tsx#L162

### What changes do you think we should make in order to solve the problem?


We should pass `includeSafeAreaPaddingBottom` as true

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?

NA
### What alternative solutions did you explore? (Optional)



**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Android - Wallet - "Get physical card" button overlaps with device navigation bar

### What is the root cause of that problem?
This happens because we have set `includeSafeAreaPaddingBottom` to false here
https://github.com/Expensify/App/blob/24793797a140b3f65a7de0511adccdcb51bc1394/src/pages/settings/Wallet/ExpensifyCardPage.tsx#L162
This is set to false because we are setting `safeAreaPaddingBottomStyle` style to the scroll view here
https://github.com/Expensify/App/blob/48b80f65e80d304965cbeb3573d7d6678fb9bd3a/src/pages/settings/Wallet/ExpensifyCardPage.tsx#L165-L171
but the all the [buttons](https://github.com/FitseTLT/App/blob/48b80f65e80d304965cbeb3573d7d6678fb9bd3a/src/pages/settings/Wallet/ExpensifyCardPage.tsx#L283-L305) are not children of the scroll view so the safe padding will not be applied for them.

### What changes do you think we should make in order to solve the problem?

The best approach is to enable `includeSafeAreaPaddingBottom` on the screen wrapper as it is the highest parent in the page. So remove this 
https://github.com/Expensify/App/blob/24793797a140b3f65a7de0511adccdcb51bc1394/src/pages/settings/Wallet/ExpensifyCardPage.tsx#L162
and then also remove `safeAreaPaddingBottomStyle` here
https://github.com/Expensify/App/blob/48b80f65e80d304965cbeb3573d7d6678fb9bd3a/src/pages/settings/Wallet/ExpensifyCardPage.tsx#L165
and render the children of the screen wrapper directly not function (as we no more need safeAreaPaddingBottomStyle) 👍  
### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?


### What alternative solutions did you explore? (Optional)

**Comment by RachCHopkins** [comment]
Yes, can repro. It's a PITA too, because when you click activate physical card, you tap the main home button  - I thought the app was crashing at first!

**Comment by mananjadhav** [comment]
@nkdengineer's [proposal](https://github.com/Expensify/App/issues/53650#issuecomment-2521263853) looks good to me.

🎀 👀 🎀 C+ reviewed.

**Comment by FitseTLT** [comment]
> @nkdengineer's [proposal](https://github.com/Expensify/App/issues/53650#issuecomment-2521263853) looks good to me.
> 
> 🎀 👀 🎀 C+ reviewed.

@mananjadhav I disagree with you. The selected proposal did not have the correct the RCA (why the [safeAreaPaddingStyle](https://github.com/Expensify/App/blob/48b80f65e80d304965cbeb3573d7d6678fb9bd3a/src/pages/settings/Wallet/ExpensifyCardPage.tsx#L165-L171)  already applied doesn't work) and the solution doesn't mention all the the needed changes that is to remove the unnecessary redundant [safeAreaPaddingStyle](https://github.com/Expensify/App/blob/48b80f65e80d304965cbeb3573d7d6678fb9bd3a/src/pages/settings/Wallet/ExpensifyCardPage.tsx#L165-L171) we are applying now and this is an important point that's why I commented a proposal in the first place. Please take a :eyes: again. Thx
cc @johnmlee101

**Comment by mananjadhav** [comment]
I think the redundant safeAreaPaddingStyle can be handled within the code. But I'll let @johnmlee101 decide.

**Comment by FitseTLT** [comment]
> I think the redundant safeAreaPaddingStyle can be handled within the code. But I'll let @johnmlee101 decide.
> 
> @johnmlee101 For you to check this [comment](https://github.com/Expensify/App/issues/53650#issuecomment-2532722576) before assigning.

It's not about removing the code in the PR phase or not but the RCA of the chosen proposal is wrong. The root cause of the issue is not even setting `includeSafeAreaPaddingBottom` to false b/c they were correct to set it to false as they have applied the  safeAreaPadding [here](https://github.com/Expensify/App/blob/48b80f65e80d304965cbeb3573d7d6678fb9bd3a/src/pages/settings/Wallet/ExpensifyCardPage.tsx#L165-L171) but the scroll view is not a parent of the buttons, this is the correct RCA and then a correct and comprehensive solution follows from the correct RCA. @johnmlee101 please take a look at my [proposal](https://github.com/Expensify/App/issues/53650#issuecomment-2524354145) correct RCA and full solution is important. Thx!

**Comment by nkdengineer** [comment]
@FitseTLT My proposal is first and your solution is just more detailed than my proposal. And the redundant safeAreaPaddingStyle can be handled in the PR phase.

**Comment by FitseTLT** [comment]
> @FitseTLT My proposal is first and your solution is just more detailed than my proposal. And the redundant safeAreaPaddingStyle can be handled in the PR phase.

Nope @nkdengineer it's not about being detailed my proposal is the only one that identifies the correct RCA and provide full solution. Let's just wait for @johnmlee101 👍

**Comment by RachCHopkins** [comment]
hey @mananjadhav @johnmlee101 I also have this [issue](https://github.com/Expensify/App/issues/53779) which appears to be the same problem in a different location (Android nav buttons sitting over Expensify buttons) - would the solution for this apply to the whole app or just this little piece?

**Comment by mananjadhav** [comment]
Looks like a similar issue and changes will be at separate places. But we can solve both of them in this same issue itself.

**Comment by mananjadhav** [comment]
Waiting for @johnmlee101 to respond on this.

**Comment by johnmlee101** [comment]
I think the RCA from @FitseTLT makes sense, and I'll go with that

**Comment by Pujan92** [comment]
@RachCHopkins @johnmlee101 @mananjadhav  Do you think we need to handle other similar issues in this single [PR](https://github.com/Expensify/App/pull/54409)?

https://github.com/Expensify/App/issues/53842
https://github.com/Expensify/App/issues/54637
https://github.com/Expensify/App/issues/53779

**Comment by mananjadhav** [comment]
Yes I agree we should handle all of them together. @johnmlee101 @FitseTLT what do you folks think?

**Comment by mountiny** [comment]
Would be great to handle this holistically cc @kirillzyusko

**Comment by mountiny** [comment]
discussing in here https://expensify.slack.com/archives/C049HHMV9SM/p1736361525995829?thread_ts=1736238279.772929&cid=C049HHMV9SM

**Comment by mananjadhav** [comment]
@RachCHopkins This is ready for payout but I think we'll need to keep the issue open because we plan to work on some follow up items.

**Comment by mananjadhav** [comment]
# BugZero Checklist:

- [x] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [x] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [x] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [x] 3d. QA
  - [ ] 3z. Other:

</details>

- [x] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment: I think we don't have a specific offending PR. We didn't consider the device navigation with buttons.

- [x] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.
 
    Link to discussion: NA

- [x] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:


Although I am suggesting the Regression Test Proposal, I think we should have one comprehensive test for the same.

## Regression Test Proposal
### Precondition:

- Device navigation is buttons and not gestures.
- User has been assigned a physical Expensify card.

### Test:


1. Go to Settings > Wallet
2. Tap on the physical card.
3. "Get physical card" / "Activate physical" button will not overlap with device navigation bar.

Do we agree 👍 or 👎

**Comment by RachCHopkins** [comment]
Sorry I have been OOO. Back today and will handle asap.

**Comment by RachCHopkins** [comment]
Payment Summary:

- Contributor: @FitseTLT to be paid $250 via Upwork
- Contributor+: @mananjadhav to be paid $250 via NewDot manual request

Upwork job [here](https://www.upwork.com/jobs/~021866268126921141487)

**Comment by RachCHopkins** [comment]
Contributor has been paid, the contract has been completed, and the Upwork post has been closed.

Leaving issue open per [comment](https://github.com/Expensify/App/issues/53650#issuecomment-2593452814) for follow-up tasks.

**Comment by garrettmknight** [comment]
$250 approved for @mananjadhav

**Comment by mananjadhav** [comment]
@mountiny Are we tracking the changes as a separate issue? Should we close this one out?

**Comment by mountiny** [comment]
Yes, the overlap/navigation bar changes are handled in a separate issue
